demosplan/DemosPlanCoreBundle/Logic/TestMailer.php

```diff
@@ -49,13 +49,13 @@ public function __construct(GlobalConfigInterface $config, private readonly Mail
         $this->emailTestTo = $config->getEmailTestTo();
     }
 
-    public function send(RawMessage $message, Envelope $envelope = null): void
+    public function send(RawMessage $message, ?Envelope $envelope = null): void
     {
         $message = $this->adjustForTestingEnvironment($message);
         $this->mailer->send($message);
     }
 
-    private function adjustForTestingEnvironment(RawMessage $message): RawMessage
+    private function adjustForTestingEnvironment(Email $message): RawMessage
```

Comment:
The change in line 58 leads to a problem in line 54 because `Email` is required but `RawMessage` is given.

Reply:
Yes, that's right. I don't know how can I fix it, because MailerInterface is a symfony component RawMessage be set there.

Reply:
To solve this it seems to be a bit more is needed than a simple type-declaration.
A possible solution could be to add a wrapping method like:
```
    /**
     * @throws TransportExceptionInterface
     */
    public function sendEmail(Email $message, ?Envelope $envelope = null): void
    {
        $this->send($this->adjustForTestingEnvironment($message), $envelope);
    }
```
and removing the line `$message = $this->adjustForTestingEnvironment($message);` in `TestMailer::send()`
and call this one  (`TestMailer::sendEmail()`) instead of `TestMailer::send()` wherever it is called.

Reply:
Unfortunatly `TestMailer::send()` be called just in vendor symfony. We can not chang it to `TestMailer::sendEmail()`

Reply:
done in [67fcb64](https://github.com/demos-europe/demosplan-core/pull/2960/commits/67fcb645c410ce4ccf2646ed27ef5c31d6e89006)